### PR TITLE
make rocksdb recovery mode configurable

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
make rocksdb recovery mode configurable

### CODE REVIEW

File: etc/config-template.toml [hunk 1]
```diff
@@ -109,6 +109,13 @@ level0-slowdown-writes-trigger = 12
 # Maximum number of level-0 files.  We stop writes at this point.
 level0-stop-writes-trigger = 16
 
+# rocksdb wal recovery mode
+# 0 : TolerateCorruptedTailRecords, tolerate incomplete record in trailing data on all logs;
+# 1 : AbsoluteConsistency, We don't expect to find any corruption in the WAL;
+# 2 : PointInTimeRecovery, Recover to point-in-time consistency;
+# 3 : SkipAnyCorruptedRecords, Recovery after a disaster;
+wal-recovery-mode = 2
```
Comment (BusyJay) [inline etc/config-template.toml]: Why not use AbsoluteConsistency?

File: src/bin/tikv-server.rs [hunk 2]
```diff
@@ -187,6 +187,15 @@ fn get_rocksdb_default_cf_option(matches: &Matches, config: &toml::Value) -> Roc
     let per_level_compression = util::config::parse_rocksdb_per_level_compression(&cpl).unwrap();
     opts.compression_per_level(&per_level_compression);
 
+    let rmode = get_integer_value("",
```
Comment (BusyJay) [inline src/bin/tikv-server.rs]: Move this to L148 seems better.

File: etc/config-template.toml (hunk 1 above)
Comment (siddontang) [inline etc/config-template.toml]: The default is 2, I think it is ok.

File: src/util/config.rs [hunk 3]
```diff
@@ -52,6 +52,16 @@ pub fn parse_rocksdb_per_level_compression(tp: &str)
     Ok(result)
 }
 
+pub fn parse_rocksdb_wal_recovery_mode(mode: i64) -> Result<DBRecoveryMode, ParseConfigError> {
```
Comment (siddontang) [inline src/util/config.rs]: add a test

**Comment by siddontang** [comment]
CI failed, seem that we should update rocksdb version in travis too. @zhangjinpeng1987

**Comment by siddontang** [comment]
Rest LGTM

**Comment by BusyJay** [comment]
Need to merge master.

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**